File: services/product-catalog-service/src/health/health.controller.ts
import { Controller, Get } from "@nestjs/common";

@Controller("health")
export class HealthController {
  @Get("live")
  live() {
    return { status: "UP" };
  }

  @Get("ready")
  ready() {
    return { status: "UP" };
  }

  @Get("startup")
  startup() {
    return { status: "UP" };
  }
}

File: services/customer-service/src/main/java/com/sales/customer/api/HealthController.java
package com.sales.customer.api;

import java.util.Map;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HealthController {
    @GetMapping("/health/live")
    public Map<String, String> live() {
        return Map.of("status", "UP");
    }

    @GetMapping("/health/ready")
    public Map<String, String> ready() {
        return Map.of("status", "UP");
    }

    @GetMapping("/health/startup")
    public Map<String, String> startup() {
        return Map.of("status", "UP");
    }
}
